Code/Framework/AzCore/Platform/Common/UnixLike/AzCore/IO/SystemFile_UnixLike.cpp

```diff
@@ -290,7 +290,11 @@ namespace AZ::IO
         }
 
         m_redirectCallback = {};
-        m_pipeData = -1;
+        if (m_pipeData > 0)
+        {
+            PosixInternal::Close(static_cast<int>(m_pipeData));
+            m_pipeData = 0;
```

Comment:
Just to be sure here, should this be 0 or -1 ?

Reply:
Actually the default value should be 0 on Windows and -1 on UnixLike platforms, so I'll move the default constructor to be platform specific `SystemFile_<Platform>.cpp` files and initialize the values accordingly.

Reply:
I updated the `FileDescriptorCapturer::Reset` function for UnixLike platforms to reset the value back to -1